### PR TITLE
Check for component container item in admin menu recovery mode

### PR DESCRIPTION
**Pull request by SniperSister**
### Summary of Changes
First, this PR reverts #13838 because it's inconsistent from a user perspective: the new admin menu has a built-in recovery mode that checks the presence of important menu items and offers a link that temporarily restores the old menu structure. With #13838 an additional check for the component container item was added - but not to the recovery mode, but to the general code of the module causing the item to show up without any explanation.

This PR moves the check for the container to the general recovery-mode logic and adjusts the messages accordingly, giving a consistent workflow.

### Testing Instructions
* Create a custom admin menu
* Set it as your active menu in menu module settings.
* A message stating that the recommended menu items for Modules, Menus and Components are missing should be shown including a link to enable the recovery mode.
* Create a new menu item of type "System Links > Components Menu Container" in your custom menu.
* The message regarding the missing component item should be gone.

### Documentation Changes Required
None

### CODE REVIEW

File: administrator/language/en-GB/en-GB.mod_menu.ini [hunk 1]
```diff
@@ -97,8 +97,9 @@ MOD_MENU_SYSTEM_INFORMATION="System Information"
 MOD_MENU_SYSTEM="System"
 MOD_MENU_TOOLS="Tools"
 MOD_MENU_USER_PROFILE="My Profile"
-MOD_MENU_WARNING_IMPORTANT_ITEMS_INACCESSIBLE="The active administrator menu does not contain links to <strong>Module Manager</strong> and <strong>Menu Manager</strong>. Click to <strong><a href='index.php?recover_menu=1'>turn on the menu recovery mode</a></strong>."
-MOD_MENU_WARNING_IMPORTANT_ITEMS_INACCESSIBLE_MENUS="The active administrator menu does not contain link to <strong>Menu Manager</strong>. Click to <strong><a href='index.php?recover_menu=1'>turn on the menu recovery mode</a></strong>."
-MOD_MENU_WARNING_IMPORTANT_ITEMS_INACCESSIBLE_MODULES="The active administrator menu does not contain link to <strong>Module Manager</strong>. Click to <strong><a href='index.php?recover_menu=1'>turn on the menu recovery mode</a></strong>."
+MOD_MENU_WARNING_IMPORTANT_ITEMS_INACCESSIBLE_MENUS="The active administrator menu does not contain link to <strong>Menu Manager</strong>."
```
Comment (dgt41) [inline administrator/language/en-GB/en-GB.mod_menu.ini]: Please change it to:
`The active administrator menu does not contain a link to`
there is a missing `a`.

Thanks Brian

**Comment by infograf768** [comment]
@izharaazmi 
what do you think?

**Comment by izharaazmi** [comment]
Two things that I don't like here:

1. The repeating "The active administrator menu does not contain a..." in all messages when we have more than one of these important items missing.
2. We are not checking any authorisation for missing "Components" menu item. Chances are their that the user has no access to any of those components and the default menu item too will hide the components menu item anyway.

Rest is fine by me. No objection to this approach. 😄

**Comment by SniperSister** [comment]
@izharaazmi: I don't get the issue you are mentioning in 2. I haven't touched the checks logic at all, just moved it to a different place?

**Comment by izharaazmi** [comment]
The menu permission is ok. What I am talking about can be reproduced as below:

1. Create a new administrator user.
2. Revoke all permission to all components under "components" menu for this new user.
3. Login as this user after setting a custom menu for the backend, which does not have components folder.
4. He still sees the warning, which he should not as it doesn't matter for him anyway.

Am I clear?

**Comment by SniperSister** [comment]
Ok, now I get what you mean - but If I read the code correctly, this won't happen anyway? The check for the container only checks if a menuitem with the type "container" exists - if so, now message is shown. Once the message processing is done, the loadItems() method is called where an empty container is hidden anyway, so this happens AFTER the check and doesn't interfere with it?

**Comment by infograf768** [comment]
@izharaazmi 
I think it is a good idea to force component container in the same way we do for modules and menu items.
Can you folks complete this patch in collaboration?

**Comment by franz-wohlkoenig** [comment]
@SniperSister is this PR to test or are you working in collaboratin like @infograf768 suggested?

**Comment by SniperSister** [comment]
I think we are waiting for @izharaazmi to provide feedback on my question

**Comment by izharaazmi** [comment]
Alright. I'll do it. :)

**Comment by franz-wohlkoenig** [comment]
Thanks for Info @SniperSister @izharaazmi

**Comment by infograf768** [comment]
@izharaazmi 
Better be fast as next release is a RC and we may freeze languages. :)

**Comment by izharaazmi** [comment]
@infograf768 I am going to show the warning against "container" only to
those users who has access to menu manager. Coz, IMO others can't do
anything about it anyway. Is that okay?

**Comment by infograf768** [comment]
i guess any custom admin menu item, including these in component container should anyway display only when the user has permissions for the components concerned.

**Comment by izharaazmi** [comment]
Please check #14235

**Comment by infograf768** [comment]
Closing in favour of https://github.com/joomla/joomla-cms/pull/14235

Thanks!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13932">issues.joomla.org/tracker/joomla-cms/13932</a>.</sub>

**Pull request closed without merging**